src/vue/vueQueryPlugin.ts

```diff
@@ -0,0 +1,53 @@
+import { isVue2 } from "vue-demi";
+import { QueryClient } from "react-query/core";
+import { VUE_QUERY_CLIENT } from "./useQueryClient";
+import { QueryClientConfig } from "./useQueryProvider";
+
+import type { Plugin } from "vue";
+
+interface ConfigOptions {
+  queryClientConfig?: QueryClientConfig;
+  queryClientKey?: string;
+}
+
+interface ClientOptions {
+  queryClient?: QueryClient;
+  queryClientKey?: string;
+}
+
+type VueQueryPluginOptions = ConfigOptions | ClientOptions;
+
+export const VueQueryPlugin: Plugin = {
+  install: (app, options: VueQueryPluginOptions = {}) => {
+    const clientKeySuffix = options.queryClientKey || "";
+    let client: QueryClient;
+
+    if ("queryClient" in options && options.queryClient) {
+      client = options.queryClient;
+    } else {
+      const clientConfig =
+        "queryClientConfig" in options ? options.queryClientConfig : undefined;
+      client = new QueryClient(clientConfig);
+    }
+
+    client.mount();
```

Comment:
I was wondering if we can track when the app is being destroyed to do unmounting.

@DamianOsipiuk  Do you have any ideas?

As I understand `mount` among other things adds event listener to do refetching on window focus. It's better to remove the listener once the app is destroyed. It can be especially important in cases when a page has multiple Vue applications (at my job we have an AngularJS app with multiple Vue apps within it)

Reply:
Relevant issue: https://github.com/vuejs/core/issues/4516

Reply:
Thanks for the link! 👍 
I there is a linked [PR](https://github.com/vuejs/core/pull/4619), hopefully it will land there soon 🤞